hermes/probe/metrics/metrics.go

```diff
@@ -0,0 +1,192 @@
+	API_GET_FILE
+)
+
+// ExitStatus represents a possible exit status metric label.
+type ExitStatus int
+
+const (
+	// SUCCESS indicates the operation/API call was successful.
+	SUCCESS ExitStatus = iota
+	// OP_TIMEOUT indicates that the operation/API call timed out.
+	OP_TIMEOUT
+	// PROBE_FAILED indicates that the probe failed due to an error.
+	PROBE_FAILED
+	// API_CALL_FAILED indicates that the API call failed due to an error.
+	API_CALL_FAILED
+	// FILE_MISSING indicates that the target file could not be found.
+	FILE_MISSING
+	// BUCKET_MISSING indicates that the target bucket could not be found.
+	BUCKET_MISSING
+	// FILE_CORRUPTED indicates that the target file was corrupted and the contents could not be read.
+	FILE_CORRUPTED
+	// FILE_READ_FAILURE indicates that the target file could not be read.
+	FILE_READ_FAILURE
+	// TODO(evanSpendlove): Review if we need this? When would we use it?
+	// FILE_METADATA_MISMATCH indicates that the target file did not match the expectation from its metadata
+	FILE_METADATA_MISMATCH
+	// UNKNOWN_FILE_FOUND indicates that an unknown file, not created by Hermes, was found in the target bucket.
+	UNKNOWN_FILE_FOUND
+	// ALL_FILES_MISSING indicates that all of the Hermes files were missing.
+	ALL_FILES_MISSING
+)
+
+var (
+	// ProbeOperations maps ProbeOperation constants to their metric label string equivalent.
+	ProbeOperations = map[ProbeOperation]string{
+		TOTAL_PROBE_RUN: "total_probe_run",
+		CHECK_NIL:       "check_nil",
+		READ_AND_VERIFY: "read_and_verify",
+		DELETE_FILE:     "delete_file",
+		CREATE_FILE:     "create_file",
+	}
+	// ApiCalls maps ApiCall constants to their metric label string equivalent.
+	ApiCalls = map[ApiCall]string{
+		API_LIST_FILES:  "list_files",
+		API_CREATE_FILE: "create_file",
+		API_DELETE_FILE: "delete_file",
+		API_GET_FILE:    "get_file",
+	}
+	// ExitStatuses maps ExitStatus constants to their metric label string equivalent.
+	ExitStatuses = map[ExitStatus]string{
+		SUCCESS:                "success",
+		OP_TIMEOUT:             "op_timeout",
+		PROBE_FAILED:           "probe_failed",
+		API_CALL_FAILED:        "api_call_failed",
+		FILE_MISSING:           "file_missing",
+		BUCKET_MISSING:         "bucket_missing",
+		FILE_CORRUPTED:         "file_corrupted",
+		FILE_READ_FAILURE:      "file_read_failure",
+		FILE_METADATA_MISMATCH: "file_metadata_mismatch",
+		UNKNOWN_FILE_FOUND:     "unknown_file_found",
+		ALL_FILES_MISSING:      "all_files_missing",
+	}
+)
+
+// Metrics stores the cumulative metrics for probe runs for a target.
+// These metrics are sent via a data channel at the end of the probe run.
+type Metrics struct {
+	StorageSystem string
+	Target        string
+	BucketName    string
+
+	// probeOpLatency is used to record latency values with distinct labels
+	// per exit status per probe operation per target.
+	// Recommended usage: probeOpLatency[probeOp][exitStatus].Metric("latency").AddFloat64(<val>)
```

Comment:
Add constants to example.
```suggestion
	// Recommended usage: probeOpLatency[ProbeOperation][ExitStatus].Metric("latency").AddFloat64(<val>)
```

Reply:
Ack.